File: backend/src/main/java/com/busticket/backend/dto/TripDTO.java
package com.busticket.backend.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class TripDTO {
    private Long id;
    private LocationDTO departureLocation;
    private LocationDTO arrivalLocation;
    private BusDTO bus;
    private LocalDateTime departureTime;
    private LocalDateTime arrivalTime;
    private BigDecimal price;
    private String duration; // From Route
    private Integer availableSeats;
}

File: app/src/main/java/com/khaiqueng_finalterm/busticketbooking/data/model/TripDTO.kt
package com.khaiqueng_finalterm.busticketbooking.data.model

import kotlinx.serialization.Serializable

@Serializable
data class TripDTO(
    val id: Long,
    val departureLocation: LocationDTO,
    val arrivalLocation: LocationDTO,
    val bus: BusDTO,
    val departureTime: String, // Dạng chuỗi ISO-8601 từ Backend
    val arrivalTime: String,
    val price: Double,
    val duration: String,
    val availableSeats: Int
)
